### DISCUSSION TITLE
a tool to convert SDK board definition into Arduino variant description

### DISCUSSION THREAD

**Discussion by lyusupov** [SiliconLabs/arduino#41] [Ideas]
It would be nice to have an automated tool to build an Arduino 'variant' descriptor from SDK definitions, when it is possible.

<img width="1128" alt="image" src="https://github.com/SiliconLabs/arduino/assets/5849637/2c0c3abb-516e-464e-9286-66c228096c34">

**Comment by silabs-bozont** [comment]
Hi,
Thank you for your suggestion!
It's not really trivial to convert an SDK supported board into an Arduino variant and ensure it'll work correctly.
We select our supported boards carefully and port them from the GSDK to Arduino while adding the necessary support code for each. Each variant/board is a bit different, you can see their SLCP files under https://github.com/SiliconLabs/arduino/tree/main/package/slcp - these are hand tailored for each board.
We aim for supporting a few boards very well instead of supporting a very wide range of them.

However in the future I think we can publish some development docs detailing the steps needed for adding a new variant from the GSDK - so if there's an individual need - then people can add support for their boards.

**Comment by lyusupov** [comment]
Thank's for your response!

The idea of the `conversion tool` was taken from **Arduino-Mbed** project.
They declare that any of Mbed-OS supported  hardware targets can be converted into Arduino `variant` folder:
https://github.com/arduino/ArduinoCore-mbed#adding-a-new-target-core-variant
